### PR TITLE
style(fix-lint-issues): fixed the lint issues

### LINKED ISSUE #19: Fix the Lint issues using the java linter
Linting is the automated checking of your source code for programmatic and stylistic errors. This is done by using a lint tool (otherwise known as a linter). A lint tool is a basic static code analyzer. Run the Linter and resolve the issues.

### PR DESCRIPTION
**Pull request by hamzashoukat94**
Used checkstyle guide for checking the linting issues and fixed the issues suggested by check style. Checkstyle can check many aspects of your source code. It can find class design problems and method design problems. It also has the ability to check code layout and formatting issues.

closes #19

### CODE REVIEW

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java [hunk 1]
```diff
@@ -51,32 +62,30 @@ public class OkClient implements HttpClient {
...
-     * @param apiLogger the logger for logging information
+     * @param compatibilityFactoryImpl the compatibilityFactory for backward compatibility
+     * @param httpLogger the logger for logging information
      */
-    public OkClient(ClientConfiguration httpClientConfig, CompatibilityFactory compatibilityFactory,
-            ApiLogger apiLogger) {
-        this(httpClientConfig, compatibilityFactory);
-        this.apiLogger = apiLogger;
+    public OkClient(final ClientConfiguration httpClientConfig,
+            final CompatibilityFactory compatibilityFactoryImpl, final ApiLogger httpLogger) {
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java:76]: `compatibilityFactoryImpl` should be reverted to `compatibilityFactory`.

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java [hunk 2]
```diff
@@ -51,32 +62,30 @@ public class OkClient implements HttpClient {
...
+     * Constructor to initialize the OKClient.
      * @param httpClientConfig the httpClientConfiguration
-     * @param compatibilityFactory the compatibilityFactory for backward compatibility
+     * @param compatibilityFactoryImpl the compatibilityFactory for backward compatibility
      */
-    public OkClient(ClientConfiguration httpClientConfig,
-            CompatibilityFactory compatibilityFactory) {
-        OkClient.compatibilityFactory = compatibilityFactory;
+    public OkClient(final ClientConfiguration httpClientConfig,
+            final CompatibilityFactory compatibilityFactoryImpl) {
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java:87]: compatibilityFactoryImpl should be reverted to compatibilityFactory.

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java [hunk 3]
```diff
@@ -51,32 +62,30 @@ public class OkClient implements HttpClient {
...
     private static CompatibilityFactory compatibilityFactory;
 
     /**
-     * Constructor to initialize the OKClient
-     * 
+     * Constructor to initialize the OKClient.
      * @param httpClientConfig client configuration
-     * @param compatibilityFactory the compatibilityFactory for backward compatibility
-     * @param apiLogger the logger for logging information
+     * @param compatibilityFactoryImpl the compatibilityFactory for backward compatibility
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java:72]: compatibilityFactoryImpl should be reverted to compatibilityFactory.

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java [hunk 4]
```diff
@@ -227,19 +235,17 @@ private long getCalculatedHeaderValue(String headerValue) {
 
     /**
      * Calculates the back-off value based on a formula which uses back-off factor and retry Count.
-     * 
      * @param requestState the current state of request entry.
      * @return long value of back-off time based on formula in milliseconds.
      */
     private long getCalculatedBackOffValue(RequestState requestState) {
-        return (long) (1000 * this.httpClientConfiguration.getRetryInterval()
+        return (long) (MILISECONDS * this.httpClientConfiguration.getRetryInterval()
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:242]: Rename MILISECONDS to TO_MILLISECOND_MULTIPLIER.

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java [hunk 5]
```diff
@@ -227,19 +235,17 @@ private long getCalculatedHeaderValue(String headerValue) {
...
                 * Math.pow(this.httpClientConfiguration.getBackOffFactor(), requestState.retryCount)
-                + Math.random() * 100);
+                + Math.random() * MAXIMUM_BACK_OFF_INTERVAL);
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:244]: Rename MAXIMUM_BACK_OFF_INTERVAL to RANDOM_NUMBER_MULTIPLIER.

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java [hunk 6]
```diff
@@ -304,9 +306,8 @@ private void logResponse(RequestState requestState, okhttp3.Response response) {
 
     /**
      * Logs the exception.
-     * 
      * @param requestState The current state of request.
-     * @param response The OKhttp Response.
+     * @param ioException the exception
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:310]: A dot is missing at the end of the description and also keep the description persistent by using the first character capital. This applies to all doc summaries.

Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:310]: Resolved

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java (hunk 5 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:244]: Done

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java (hunk 4 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:242]: Done

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java (hunk 3 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java]: Reverted

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java [hunk 7]
```diff
@@ -51,32 +62,30 @@ public class OkClient implements HttpClient {
...
+        this.apiLogger = httpLogger;
     }
 
     /**
-     * Constructor to initialize the OKClient
-     * 
+     * Constructor to initialize the OKClient.
      * @param httpClientConfig the httpClientConfiguration
-     * @param compatibilityFactory the compatibilityFactory for backward compatibility
+     * @param compatibilityFactoryImpl the compatibilityFactory for backward compatibility
```
[the remark this answers is not in the document]
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java]: Reverted

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java (hunk 2 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java]: Reverted

File: src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java (hunk 1 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/OkClient.java]: Reverted

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java (hunk 6 above)
Comment (sufyankhanrao) [reply src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java]: Are there other occurrences in the project? if yes, kindly resolve them.

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java [hunk 8]
```diff
@@ -23,6 +23,20 @@
  */
 public class RetryInterceptor implements Interceptor {
 
+    /**
+     * Maximum Back off interval.
+     */
+    private static final int RANDOM_NUMBER_MULTIPLIER = 100;
+
+    /**
+     * Maximu Retry interval.
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java:32]: A Typo on this line.

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java (hunk 6 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java]: Resolved

File: src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java (hunk 8 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/okhttpclient/adapter/interceptors/RetryInterceptor.java]: Fixed

**Review by sufyankhanrao — APPROVED** [review]

**Pull request merged**